### PR TITLE
real trading fixes

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: tests/data/test_portfolio_profitability.py [hunk 1]
```diff
@@ -122,50 +121,56 @@ async def test_get_current_crypto_currencies_values(backtesting_trader):
...
-    await portfolio_profitability.handle_mark_price_update("ETH/BTC", eth_btc_price)
-    assert await portfolio_profitability.get_current_crypto_currencies_values() == {
-        'BTC': 1,
-        'ETH': eth_btc_price,
-        'XRP': 0.005,
-        'NANO': 0.05,
-        'XLM': xlm_btc_price,
-        'USDT': 1 / usdt_btc_price
-    }
+    if not os.getenv('CYTHON_IGNORE'):
```
Comment (Herklos) [inline tests/data/test_portfolio_profitability.py:124]: :disappointed_relieved:

Comment (Herklos) [reply tests/data/test_portfolio_profitability.py:124]: Due to `missing_currency_data_in_exchange` ?

Comment (GuillaumeDSM) [reply tests/data/test_portfolio_profitability.py:124]: Due to it being private yes

File: octobot_trading/exchanges/exchange_manager.py [hunk 2]
```diff
@@ -300,6 +299,7 @@ async def _create_exchange_producers(self):
 
     def _is_managed_by_websocket(self, channel):  # TODO improve checker
         return not self.rest_only and self.has_websocket and \
+               channel in WEBSOCKET_FEEDS_TO_TRADING_CHANNELS and \
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_manager.py:302]: :+1:

File: octobot_trading/producers/balance_updater.py [hunk 3]
```diff
@@ -82,7 +82,7 @@ class BalanceProfitabilityUpdater(BalanceProfitabilityProducer):
     """
     The updater related channel name
     """
-    CHANNEL_NAME = BALANCE_CHANNEL
+    CHANNEL_NAME = BALANCE_PROFITABILITY_CHANNEL
```
Comment (Herklos) [inline octobot_trading/producers/balance_updater.py:85]: :+1:

File: octobot_trading/data_manager/portfolio_manager.py [hunk 4]
```diff
@@ -52,15 +52,19 @@ def handle_balance_update(self, balance):
             return self.portfolio.update_portfolio_from_balance(balance)
         return False
 
-    def handle_balance_update_from_order(self, order):
+    async def handle_balance_update_from_order(self, order):
```
Comment (Herklos) [inline octobot_trading/data_manager/portfolio_manager.py:55]: async ?

Comment (GuillaumeDSM) [reply octobot_trading/data_manager/portfolio_manager.py:55]: await self.exchange_manager.exchange.get_balance()

Comment (Herklos) [reply octobot_trading/data_manager/portfolio_manager.py:55]: oh yes my bad

File: octobot_trading/data_manager/portfolio_manager.py [hunk 5]
```diff
@@ -52,15 +52,19 @@ def handle_balance_update(self, balance):
...
-        if self.trader.is_enabled and self.trader.simulate:
-            self.portfolio.update_portfolio_from_order(order)
-            return True
+        if self.trader.is_enabled:
+            if self.trader.simulate:
+                self.portfolio.update_portfolio_from_order(order)
+                return True
+            else:
+                # on real trading: reload portfolio to ensure portfolio sync
+                return self.handle_balance_update(await self.exchange_manager.exchange.get_balance())
```
Comment (Herklos) [inline octobot_trading/data_manager/portfolio_manager.py:67]: It's not something we want to do here

Comment (Herklos) [reply octobot_trading/data_manager/portfolio_manager.py:67]: We wanted to request exchange only in updaters in v4

File: octobot_trading/channels/balance.py [hunk 6]
```diff
@@ -45,6 +45,16 @@ async def send(self, balance):
                 "balance": balance
             })
 
+    async def update_portfolio_from_exchange(self, should_notify=False) -> bool:
```
Comment (Herklos) [inline octobot_trading/channels/balance.py:48]: :+1:

**Pull request merged**